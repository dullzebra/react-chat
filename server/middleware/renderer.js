import React from 'react';
import ReactDOMServer from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import { SheetsRegistry } from 'react-jss/lib/jss';
import JssProvider from 'react-jss/lib/JssProvider';
import { createGenerateClassName } from 'material-ui/styles';

import configureStore from '../../src/store';
import App from '../../src/components/App';

const sheetsRegistry = new SheetsRegistry();
const generateClassName = createGenerateClassName();

const path = require('path');
const fs = require('fs');

export default (req, res) => {
  // point to the html file created by CRA's build tool
  const filePath = path.resolve(__dirname, '..', '..', 'build', 'index.html');
  const context = {};
  const store = configureStore();

  fs.readFile(filePath, 'utf8', (err, htmlData) => {
    if (err) {
      console.error('err', err);
      return res.status(404).end();
    }

    if (context.url) {
      res.redirect(context.url);
      return;
    }

    // render the app as a string
    /* eslint-disable-next-line */
    const html = ReactDOMServer.renderToString(
      <JssProvider registry={sheetsRegistry} generateClassName={generateClassName}>
        <Provider store={store}>
          <StaticRouter location={req.url} context={context}>
            <App />
          </StaticRouter>
        </Provider>
      </JssProvider>);

    const css = sheetsRegistry.toString();

    // inject the rendered app into our html and send it
    return res.end(htmlData.replace(
      '<div id="root"></div>',
      `<div id="root">${html}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState()).replace(
    /</g,
    '\\u003c',
  )}</script>
    <style id="jss-server-side">${css}</style>
    `,
    ));
  });
};
