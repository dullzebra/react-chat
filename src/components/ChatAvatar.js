import React from 'react';
import Avatar from 'material-ui/Avatar';
import initials from '../utils/initials';
import getColor from '../utils/color-from';

export default function ChatAvatar({ name }) {
  return <Avatar style={{backgroundColor: getColor(initials(name))}}>{initials(name)}</Avatar>
}
