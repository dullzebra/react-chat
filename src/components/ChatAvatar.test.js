/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatAvatar from './ChatAvatar';

describe('<ChatAvatar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatAvatar name="Piggy Chat Title" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatAvatar name="Piggy Chat Title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
