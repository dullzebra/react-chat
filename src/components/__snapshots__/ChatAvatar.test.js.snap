// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatAvatar /> renders correctly 1`] = `
<div
  className="MuiAvatar-root-4 MuiAvatar-colorDefault-5"
  style={
    Object {
      "backgroundColor": "#9c27b0",
    }
  }
>
  PC
</div>
`;
